src/services/transfer/transfer-out-util.js

```diff
@@ -51,70 +52,32 @@ export const updateFragmentStatus = async (conversationId, messageId, status) =>
   logInfo(`Updated fragment status to: ${status}`);
 };
 
-const replaceMessageIdInObject = (ehrMessage, oldMessageId, newMessageId) => {
-  // helper func to replace messageIds in an EHR core or EHR fragment parsed as a JS object
+export const replaceMessageIdsInObject = (ehrMessage, messageIdReplacements) => {
+  let ehrMessageJsonString = JSON.stringify(ehrMessage);
 
-  return JSON.parse(JSON.stringify(ehrMessage).replaceAll(oldMessageId, newMessageId));
-};
+  messageIdReplacements.forEach(messageIdReplacement => {
+    ehrMessageJsonString =
+        ehrMessageJsonString.replaceAll(messageIdReplacement.oldMessageId, messageIdReplacement.newMessageId);
+  });
 
-export const updateMessageIdForEhrCore = async ehrCore => {
-  try {
-    const messageId = await parseMessageId(ehrCore);
-    const newMessageId = uuidv4().toUpperCase();
-    const ehrCoreWithUpdatedMessageId = replaceMessageIdInObject(ehrCore, messageId, newMessageId);
+  return JSON.parse(ehrMessageJsonString);
+}
 
+export const createNewMessageIds = async oldMessageIds => {
+  const messageIdReplacements = oldMessageIds.map((oldMessageId) => {
     return {
-      newMessageId, ehrCoreWithUpdatedMessageId
+      oldMessageId: oldMessageId,
+      newMessageId: uuidv4().toUpperCase()
     }
-  } catch (error) {
-    throw new MessageIdUpdateError(error);
-  }
-};
+  });
 
-export const createNewMessageIdsForAllFragments = async fragmentMessageIds => {
-  for (let oldMessageId of fragmentMessageIds) {
-    const newMessageId = uuidv4().toUpperCase();
-    await createMessageIdReplacement(oldMessageId, newMessageId);
-  }
+  await createMessageIdReplacements(messageIdReplacements);
+  logInfo(`Created new message ID(s) for EHR core & ${messageIdReplacements.length - 1} fragments`);

```

Comment:
What if there was 1 fragment, would this be `Created new message ID(s) for EHR core & 0 fragments`?